odl/solvers/util/callback.py

```diff
@@ -438,19 +440,137 @@ def reset(self):
         """Set `iter` to 0 and create a new figure."""
         self.iter = 0
         self.fig = None
+        self.space_of_last_x = None
```

Comment:
You could theoretically use a callback with several solvers, we even have the whole "reset" functionality in order to support this use case.